stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/querychannels/WhenQuery.kt

```diff
@@ -0,0 +1,44 @@
+package io.getstream.chat.android.offline.querychannels
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.livedata.ChatDomainImpl
+import io.getstream.chat.android.test.TestCall
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+@ExperimentalCoroutinesApi
+internal class WhenQuery {
+    private lateinit var sut: QueryChannelsController
+    private lateinit var chatClient: ChatClient
+
+    @BeforeEach
+    fun before() {
+        val chatDomainImpl = mock<ChatDomainImpl> {
+            on { scope } doReturn TestCoroutineScope()
+            on { repos } doReturn mock()
+        }
+        chatClient = mock()
+        sut = QueryChannelsController(mock(), mock(), chatClient, chatDomainImpl)
+    }
+
+    @Test
+    fun `Given already loading state Should return error for query`() = runBlockingTest {
+        pauseDispatcher {
+            whenever(chatClient.queryChannels(any())) doReturn TestCall(Result(emptyList()))
+            sut.query()
+
+            val result = sut.query()
+
+            // TODO should return error
+            // Truth.assertThat(result.isError).isTrue()
```

Comment:
Is this still in progress?

Reply:
Yes, I'm working on it. Wants to merge to avoid conflicts